Sources/SwiftParser/Declarations.swift

```diff
@@ -167,7 +167,13 @@ extension Parser {
           arena: parser.arena)
       } addSemicolonIfNeeded: { lastElement, newItemAtStartOfLine, parser in
         if lastElement.semicolon == nil && !newItemAtStartOfLine {
-          return lastElement.withSemicolon(parser.missingToken(.semicolon, text: nil), arena: parser.arena)
```

Comment:
Totally fine with this change for now. But one thing I was thinking of... What if we instead had copy initializers that take the previous node + optional params + leading/trailing trivia? For both the red + green tree. Then this would be eg.
```
return RawMemberDeclListItemSyntax(lastElement, semicolon: parser.missingToken(.semicolon, text: nil)))
```